alf/algorithms/data_transformer.py

```diff
@@ -122,17 +122,24 @@ def __init__(self, data_transformer_ctors, observation_spec):
         """
         data_transformers = nn.ModuleList()
         has_non_frame_stacker = False
+        has_non_hindsight = False
         state_spec = []
         max_stack_size = 1
         for ctor in data_transformer_ctors:
             obs_trans = ctor(observation_spec)
             if isinstance(obs_trans, FrameStacker):
                 max_stack_size = max(max_stack_size, obs_trans.stack_size)
                 assert not has_non_frame_stacker, (
-                    "FrameStacker need to be the "
-                    "first data transformers if it is used.")
-            elif not isinstance(obs_trans, HindsightExperienceTransformer):
+                    "FrameStacker needs to be the "
+                    "first data transformer if it is used.")
+            else:
                 has_non_frame_stacker = True
+            if isinstance(obs_trans, HindsightExperienceTransformer):
+                assert not has_non_hindsight, (
+                    "HindsightExperienceTransformer needs to be the "
+                    "first data transformer if it is used.")
+            else:
+                has_non_hindsight = True
```

Comment:
The logic here is right. But can be frustrating if a user tries switching the order of the two and finds both need to be the first. Should assert explicitly that they don't co-exist.

Reply:
Good point.  Simplified a bit.  How about now?